docs/binding-controller.md

```diff
@@ -0,0 +1,60 @@
+```mermaid
+%%{init: {'theme':'forest'}}%%
+flowchart TD
+  %% Defining the styles
+    classDef Red fill:#FF9999;
+    classDef Amber fill:#FFDEAD;
+    classDef Green fill:#BDFFA4;
+
+  predicate((PREDICATE: <br>SnapshotEnvironmentBinding<br>is created or updated))
+
+%%%%%%%%%%%%%%%%%%%%%%% Drawing EnsureIntegrationTestPipelineForScenarioExists() function
+
+%% Node definitions
+ensure1(Proceed further if:<br>Snapshot testing <br> not finished yet)
+isThereAnITS("Is there an<br>IntegrationTestScenario<br>present in the binding?")
+continueProcessing1[/Controller continues processing.../]
+getLatestPipelineRun("Get latest pipeline run for<br>snapshot and scenario")
+isPipelineRunExisting("Does an integration<br>pipeline run exist?")
+createNewPipelineRun("Create a new pipeline run<br>for the snapshot")
+
+
+%% Node connections
+predicate               ----> |"EnsureIntegrationTestPipelineForScenarioExists()"|ensure1
+ensure1               ---->       isThereAnITS
+isThereAnITS            --No-->     continueProcessing1
+isThereAnITS            --Yes-->    getLatestPipelineRun
+getLatestPipelineRun    ---->       isPipelineRunExisting
+isPipelineRunExisting   --No-->     continueProcessing1
+isPipelineRunExisting   --Yes-->    createNewPipelineRun
+createNewPipelineRun    ---->       continueProcessing1
+
+%%%%%%%%%%%%%%%%%%%%%%% Drawing createIntegrationPipelineRunWithEnvironment() function
+
+%% Node definitions
+getTarget(Get deployment target for<br>environment)
+createPipelineRun(Create new pipelineRun for<br>application/snapshot/environment combination)
+continueProcessing2[/Controller continues processing.../]
+
+
+%% Node connections
+predicate         ---->   |"createIntegrationPipelineRunWithEnvironment()"|getTarget
```

Comment:
I find this link between this function and predicate a little confusing. It makes it seem like the controller calls this function directly, which is not the case. I think we should add this branch within the `EnsureIntegrationTestPipelineForScenarioExists()` branch where it gets. Maybe something like this:


```mermaid
graph TD;
isPipelineRunExisting("Does an integration<br>pipeline run exist?")
getDeploymentTargetClaim(Load deployment target claim<br>from cluster using<br>environment namespace)
getDeploymentTarget(Load deployment target<br>from cluster using<br>deployment target claim)
createPipelineRun(Create new pipelineRun for<br>application/snapshot/environment combination)
continueProcessing3[/Controller continues processing.../]

isPipelineRunExisting   --No-->    getDeploymentTargetClaim
getDeploymentTargetClaim -->   getDeploymentTarget
getDeploymentTarget -->  createPipelineRun
createPipelineRun --> continueProcessing3
```

I basically shifted the content of the first 2 branches here ^
wdyt?

Reply:
I'm going to circle back to this once we figure out how to phrase the predicate.  That may impact what we do with the items that follow.

Reply:
I think that makes sense.  We should probably do something similar with `getDeploymentTargetForEnvironment()`.  That being said, since this flowchart details how the controller itself works there's an argument to be made that only the `Ensure...()` functions should be diagrammed.  That's also in line with the way the already-documented controller flowcharts look.

Reply:
That is a good point - I'm not sure if individual functions need to be branched out here because e.g. `getDeploymentTargetForEnvironment()` is a prerequisite fact-gathering function that sets the stage for further `Ensure...` functions.

Reply:
Sounds like we have consensus.  I'll remove those functions